pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/bestpractices/xml/DebugsShouldSpecifyLoggingLevel.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data
+        xmlns="http://pmd.sourceforge.net/rule-tests"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests http://pmd.sourceforge.net/rule-tests_1_0_0.xsd">
+
+    <test-code>
+        <description>Problematic apex debug statements - System.debug should have two arguments.</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>5</expected-linenumbers>
+        <code><![CDATA[
+@isTest
+public class Foo {
+    @isTest
+    static void methodATest() {
+        System.debug('I am a bad debug!'); // not good
+        System.debug(LoggingLevel.WARN, 'I am a good warning.') // good
+    }
+}
+		]]></code>
+    </test-code>
+
+    <test-code>
+        <description>When using strict mode, one should not use the DEBUG LoggingLevel.</description>
+        <rule-property name="strictMode" value="true"/>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>5</expected-linenumbers>
+        <code><![CDATA[
+@isTest
+public class Foo {
+    @isTest
+    static void methodATest() {
+        System.debug(LoggingLevel.DEBUG, 'I am a bad debug.'); // not good
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>When not using strict mode (default), one can use the DEBUG LoggingLevel.</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+@isTest
+public class Foo {
+    @isTest
+    static void methodATest() {
+        System.debug(LoggingLevel.DEBUG, 'I am a good debug.') // good
+    }
```

Comment:
Are there other methods on the System class? We should add them here, to avoid we don't flag other methods, that we shouldn't...

Reply:
Yes there are many other methods in system. So we need to fully qualify it.

Reply:
Yes there are many other methods in system. So we need to fully qualify it.